### PR TITLE
Add word2vec

### PR DESCRIPTION
**Pull request by jacquesqiao**

### CODE REVIEW

File: fluid/PaddleRec/word2vec/data/download.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+wget http://mattmahoney.net/dc/enwik8.zip
```
Comment (seiriosPlus) [inline fluid/PaddleRec/word2vec/data/download.sh]: Need put the data on BOS?

File: fluid/PaddleRec/word2vec/train.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        '--is_local',
+        type=int,
+        default=1,
+        help='Local train or distributed train (default: 1)')
+    # the following arguments is used for distributed train, if is_local == false, then you should set them
+    parser.add_argument(
+        '--role',
+        type=str,
+        default='pserver',  # trainer or pserver
+        help='The path for model to store (default: models)')
```
Comment (seiriosPlus) [inline fluid/PaddleRec/word2vec/train.py]: The introduction error.

File: fluid/PaddleRec/word2vec/train.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    parser.add_argument(
+        '--current_endpoint',
+        type=str,
+        default='127.0.0.1:6000',
+        help='The path for model to store (default: 127.0.0.1:6000)')
+    parser.add_argument(
+        '--trainer_id',
+        type=int,
+        default=0,
+        help='The path for model to store (default: models)')
```
Comment (seiriosPlus) [inline fluid/PaddleRec/word2vec/train.py]: Error too.

File: fluid/PaddleRec/word2vec/train.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        train_loop(args, main_program, word2vec_reader, data_list, loss, 1, -1)
+    else:
+        logger.info("run dist training")
+        t = fluid.DistributeTranspiler()
+        t.transpile(
+            args.trainer_id, pservers=args.endpoints, trainers=args.trainers)
+        if args.role == "pserver":
+            logger.info("run pserver")
+            prog = t.get_pserver_program(args.current_endpoint)
+            startup = t.get_startup_program(
```
Comment (seiriosPlus) [inline fluid/PaddleRec/word2vec/train.py:303]: Use ```get_pserver_programs``` ?

File: fluid/PaddleRec/word2vec/cluster_train.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+# start trainer1
+python train.py \
+    --train_data_path /paddle/data/train.txt \
+    --is_local 0 \
+    --role trainer \
+    --endpoints 127.0.0.1:6000,127.0.0.1:6001 \
+    --trainers 2 \
+    --trainer_id 1 \
+    > trainer1.log 2>&1 &
```
Comment (seiriosPlus) [inline fluid/PaddleRec/word2vec/cluster_train.sh]: A new line at the end.

File: fluid/PaddleRec/word2vec/README.cn.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+```bash
+sh cluster_train.sh
+```
+
+## 预测
+
+
+## 在百度云上运行集群训练
+1. 参考文档 [在百度云上启动Fluid分布式训练](https://github.com/PaddlePaddle/FluidDoc/blob/develop/doc/fluid/user_guides/howto/training/train_on_baidu_cloud_cn.rst) 在百度云上部署一个CPU集群。
+1. 用preprocess.py处理训练数据生成train.txt。
```
Comment (seiriosPlus) [inline fluid/PaddleRec/word2vec/README.cn.md:74]: 序号错误.

Comment (jacquesqiao) [reply fluid/PaddleRec/word2vec/README.cn.md:74]: 这个序号是对的，markdown会作自动编号

File: fluid/PaddleRec/word2vec/reader.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+        if not with_hs:
+            return _reader
+        else:
+            return _reader_hs
+
+
+if __name__ == "__main__":
+    epochs = 10
+    batch_size = 1000
```
Comment (guru4elephant) [inline fluid/PaddleRec/word2vec/reader.py]: it seems batch size is window num, could you add comment here?

**Review by guru4elephant — COMMENTED** [review]
meaning of batch_size should have comments

**Review by guru4elephant — COMMENTED** [review]
please add some comments

File: fluid/PaddleRec/word2vec/README.cn.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+对数据进行预处理以生成一个词典。
+
+```bash
+python preprocess.py --data_path ./data/1-billion-word-language-modeling-benchmark-r13output/training-monolingual.tokenized.shuffled --dict_path data/1-billion_dict
+```
+
+## 训练
+训练的命令行选项可以通过`python train.py -h`列出。
+
+### 单机训练：
```
Comment (jacquesqiao) [inline fluid/PaddleRec/word2vec/README.cn.md:32]: add doc for inference

**Review by jacquesqiao — COMMENTED** [review]
LGTM!

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**